### PR TITLE
test: Allow minor polygon differences

### PR DESCRIPTION
**Pull request by l0b0**
### Motivation

Make the test pass in both GDAL 3.8.5 and 3.9.2, without reducing the accuracy of the assertion significantly.

### Modifications

Change the polygon tolerance from exactly zero to 1 part in 10 quadrillion.

### Verification

`pytest`

### CODE REVIEW

File: scripts/stac/imagery/tests/capture_area_test.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def test_merge_polygons_with_rounding_margin_too_big() -> None:
     print(f"GeoJSON expected: {to_feature(expected_merged_polygon)}")
     print(f"GeoJSON result: {to_feature(merged_polygons)}")
 
-    assert merged_polygons.equals_exact(expected_merged_polygon, tolerance=0.0)
+    assert merged_polygons.equals_exact(expected_merged_polygon, tolerance=0.000_000_000_000_000_1)
```
Comment (blacha) [inline scripts/stac/imagery/tests/capture_area_test.py:73]: where did 1e-16 come from? isn't that much less than the accuracy of a float (or are these stored as doubles?)

for our usecases on lat,lon anything less than 1e-10 is well outside our margin of error, in other places we just hard code a error margin to something like Epsilon?

Comment (l0b0) [reply scripts/stac/imagery/tests/capture_area_test.py:73]: It was the smallest tolerance (I only tested multiples of 10) which made the test pass on both versions of GDAL.

Comment (blacha) [reply scripts/stac/imagery/tests/capture_area_test.py:73]: I would prefer we choose a precision that is relevant to us eg 1e-10 1/100th mm or 1e-12 and store it as a rounding const somewhere, so then when we run into more floating point math rounding issues we can just reference the const.

We could also use epslion (~2.2e-16) Not sure how you get it in python in the browser you can use `Number.EPSLION`

Comment (l0b0) [reply scripts/stac/imagery/tests/capture_area_test.py:73]: Epsilon is usually the smallest representable number of the relevant type, so it's pretty much guaranteed to be too small.

Are the units of these polygons always in metres? If so, something like 1e-6 (1 micrometre) should be plenty. If it's anything big, like degrees, it's going to be much more difficult to come up with a specific value which is sufficiently precise.

Comment (blacha) [reply scripts/stac/imagery/tests/capture_area_test.py:73]: degrees map pretty generally to meters at a 111KM per degree level, so 8DP of degrees is 1mm at the equator and slightly less in New Zealand which is somewhere near 80KM per degree 

ref: https://en.wikipedia.org/wiki/Decimal_degrees

> Epsilon is usually the smallest representable number of the relevant type, so it's pretty much guaranteed to be too small.

1e-16 is smaller than Epsilon.

```typescript
0.000_000_000_000_000_1 < Number.EPSILON // true
```

Comment (l0b0) [reply scripts/stac/imagery/tests/capture_area_test.py:73]: Looks like I misunderstood how that works. Implementing…

**Review by blacha — APPROVED** [review]

**Pull request merged**